pkg/api/labels.go

```diff
@@ -0,0 +1,56 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/NYTimes/gziphandler"
+	"github.com/prometheus/prometheus/promql/parser"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/timescale/timescale-prometheus/pkg/promql"
+	"github.com/timescale/timescale-prometheus/pkg/query"
+	"math"
+	"net/http"
+	"strings"
+)
+
+type labelsValue []string
+
+func (l labelsValue) Type() parser.ValueType {
+	return parser.ValueTypeNone
+}
+
+func (l labelsValue) String() string {
+	return strings.Join(l, "\n")
+}
+
+func Labels(queriable *query.Queryable) http.Handler {
+	hf := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Access-Control-Allow-Origin", "*")
```

Comment:
This is controlled by a command line flag in Prom 
```
	a.Flag("web.cors.origin", `Regex for CORS origin. It is fully anchored. Example: 'https?://(domain1|domain2)\.com'`).
		Default(".*").StringVar(&cfg.corsRegexString)
```

Should we implement this in the future too (if so please file an issue)

Reply:
Also why isn't this in `setHeaders`?

Reply:
Moved to setHeaders, and created a card in project board